org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/event/EventRecordingAssert.java

```diff
@@ -0,0 +1,106 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.assertj.event;
+
+import java.util.function.Predicate;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.FrameworkEvent;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.test.common.event.EventRecording;
+
+public class EventRecordingAssert extends AbstractEventRecordingAssert<EventRecordingAssert, EventRecording> {
+
+	@SuppressWarnings("unchecked")
+	private static <T> Predicate<Object> hasTypeAnd(Class<T> clazz, Predicate<T> predicate) {
+		return e -> clazz.isAssignableFrom(e.getClass()) && predicate.test((T) e);
+	}
+
+	public EventRecordingAssert(EventRecording actual, Class<EventRecordingAssert> selfType) {
+		super(actual, selfType);
+	}
+
+	public static EventRecordingAssert assertThat(EventRecording actual) {
+		return new EventRecordingAssert(actual, EventRecordingAssert.class);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the
+	 *            ServiceEvents
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThatServiceEvent(Runnable execute, Predicate<ServiceEvent> predicate,
+		int timeout) throws Exception {
+		return assertThat(execute, hasTypeAnd(ServiceEvent.class, predicate), timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the
+	 *            BundleEvents
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThatBundleEvent(Runnable execute, Predicate<BundleEvent> predicate,
+		int timeout) throws Exception {
+		return assertThat(execute, hasTypeAnd(BundleEvent.class, predicate), timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the
+	 *            FrameworkEvents
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThatFrameworkEvent(Runnable execute, Predicate<FrameworkEvent> predicate,
+		int timeout) throws Exception {
+		return assertThat(execute, hasTypeAnd(FrameworkEvent.class, predicate), timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the Events
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThat(Runnable execute, Predicate<Object> predicate, int timeout)
```

Comment:
The predicate argument should be typed `Predicate<?>` if you don't care what the type for the predicate input is. Using `Predicate<Object>` means that people cannot easily using a method reference to an existing method whose argument is not Object.

Reply:
I used `Predicate<? super Object>` because `Predicate<?>` will result in a compilation error on EventRecording.record.
```
The method test(capture#13-of ?) in the type Predicate<capture#13-of ?> is not applicable for the arguments (BundleEvent)	EventRecording.java
```

is this OK for you?

Reply:
I don't think `Predicate<? super Object>` is helpful. I can't pass a `Predicate<BundleEvent>` argument. YOu may need some casting in the method?

Reply:
thats my last idea;
https://github.com/osgi/osgi-test/pull/257/commits/36aea8fff80fc53063f0e3180940b37cba7d139f